File: test/all_cover.pas
Program example(input, output);

Const 
    a = 5; 
    b = -6; 
    c = 't';
    
Var 
    x, y, d : integer;
    e : array [1..50] of char;
    f : record
        f1 : real;
        f2 : array [1..50, 1..100] of boolean;
        f3 : record
            f31 : real;
            f32 : array [1..50, 1..100] of boolean;
            f33 : record
                f331 : real;
                f332 : array [1..50, 1..100] of boolean;
                f333 : integer;
            End;
        End;
    End;
    g : array [1..50, 1..100] of boolean;

Function func(var a: integer; b:real) : integer;
// Function func(var a : integer,b : integer) : integer;
var i:integer;
Begin
    for i:=1 to a do
        begin
          write(b);
        end;
    While i<5 Do
        begin
            write(y);
        end;
    func(a,b);
    If b = 0 Then func := a
    Else func := func(a, b);
End;

Begin
    read(x,y);
    write(x,y);
    while x<>y do write(x,y);
    func(x, y)
End.

File: test/all_cover.c
#include<stdbool.h>
#include<stdio.h>

const int a = 5; 
const int b = -6; 
const char c = 't'; 

int x, y, d; 
char e[50]; 
struct {
	float f1; 
	bool f2[50][100]; 
	struct {
		float f31; 
		bool f32[50][100]; 
		struct {
			float f331; 
			bool f332[50][100]; 
			int f333; 
		}
		f33; 
	}
	f3; 
}
f; 
bool g[50][100]; 

int func(int* a, float b){
	int i; 
	
	if(b == 0){
		return *a; 
	}
	else{
		return func(a, b); 
	}
	for(i = 1; i < *a; i++){
		printf("b: %f\n", b); 
	}
	i = 0; 
	while(i != 5){
		printf("b: %f\n", b); 
		i = i + 1; 
	}
	func(a, b); 
}

int main(int argc, char* argv[]){
	printf("f.f3.f31: %f\n", f.f3.f31); 
	scanf("%d%d", &x, &y); 
	printf("x: %d\ny: %d\n", x, y); 
	func(&x, y); 
}
